test/counter_wintf_test.dart

```diff
@@ -32,7 +34,9 @@ class CounterInterface extends Interface<CounterDirection> {
 
 class Counter extends Module {
   late final CounterInterface intf;
-  Counter(CounterInterface intf) {
+  late bool resetRoot;
+  Counter(CounterInterface intf,
+      {this.resetRoot = false, bool withResetVal = false}) {
```

Comment:
It's a little confusing how you have this, and unclear what `resetRoot` means.  There's 2 different things you can control:
- Should it be reset the "old" way, or the new way you implemented
- What should the reset value be?

It might be clearer to have those two arguments in the constructor for this `Counter` instead:
- `bool useBuiltInSequentialReset`: if `true`, then use the new way, otherwise use the old way
- `int resetValue`: the value that should be reset, regardless of the mode

Then you can use a single if/else clause based on `useBuiltInSequentialReset`, and in each of the two cases use `resetValue` to reset it.  What do you think?